app/controllers/shipit/ccmenu_url_controller.rb

```diff
@@ -0,0 +1,22 @@
+require 'uri'
+
+module Shipit
+  class CcmenuUrlController < ShipitController
+    def fetch
+      uri = URI(api_stack_ccmenu_url(stack_id: stack.to_param))
+      uri.query = {'token' => client.authentication_token}.to_query
+      render json: {ccmenu_url: uri.to_s}
+    end
+
+    private
+
+    def client
+      @client ||= Shipit::ApiClient.create_with(permissions: %w(read:stack))
```

Comment:
Nitpick, since you are in the `Shipit` namespace alreasy, you can reference `ApiClient` directly.